src/VisualStudio/Core/Def/Implementation/Options/LocalUserRegistryOptionPersister.cs

```diff
@@ -30,12 +28,10 @@ internal sealed class LocalUserRegistryOptionPersister : IOptionPersister
         private readonly object _gate = new();
         private readonly RegistryKey _registryKey;
 
-        [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public LocalUserRegistryOptionPersister([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+        public LocalUserRegistryOptionPersister(IThreadingContext threadingContext, IServiceProvider serviceProvider)
         {
-            // Starting in Dev16, the ILocalRegistry service behind this call is free-threaded, and since the service is offered by msenv.dll can be requested
-            // without any marshalling (explicit or otherwise) to the UI thread.
+            threadingContext.ThrowIfNotOnUIThread();
```

Comment:
Why add this assert? Was the earlier comment incorrect?

Reply:
the comment was not supported by [public documentation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.ilocalregistry?view=visualstudiosdk-2019), so I removed it

Reply:
Also note that the implementation itself is not intended for use on background threads:
https://devdiv.visualstudio.com/DevDiv/_git/VS?path=%2Fsrc%2Fenv%2Fshell%2FPackageFramework%2FCurrent%2FShell%2FVSRegistry.cs&version=GBmain&line=63&lineEnd=64&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Reply:
I think the trick with this one is it's special cased:

https://devdiv.visualstudio.com/DevDiv/_git/VS?path=%2Fsrc%2Fenv%2Fmsenv%2Fcore%2Fproffer.cpp&version=GBmain&line=766&lineEnd=775&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

https://devdiv.visualstudio.com/DevDiv/_git/VS?path=%2Fsrc%2Fenv%2Fmsenv%2Fcore%2Fproffer.cpp&version=GBmain&line=1737&lineEnd=1738&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Reply:
This is why https://github.com/dotnet/roslyn/pull/52063#discussion_r599773974 is the only sane approach to handling this...

Reply:
As far as I'm concerned, the comment was the public documentation. :smile:

Reply:
I don't mind the assert now being there given the larger context is now doing the call on the UI thread, but can we keep the comment? Or revise it to explain why the assert is there? I expect this to move around further and having to go through this again would be unfortunate.

Reply:
Documentation is now more complete

Reply:
the assert feels weird to me.  this was made free threaded and communicated to us that way.  if we think their documentation should be more sufficient, i woudl prefer we just ask them to do that.